public/app/features/dashboard-scene/scene/DashboardScene.tsx

```diff
@@ -407,6 +407,24 @@ export class DashboardScene extends SceneObjectBase<DashboardSceneState> {
     return this._initialState;
   }
 
+  public addPanel(vizPanel: VizPanel): void {
```

Comment:
@mdvictor, I've created this method that only handles adding when dashboard is empty. We can work together on non-empty dashboard logic